pushapkscript/test/helpers/mock_file.py

```diff
@@ -0,0 +1,19 @@
+class MockFile:
```

Comment:
You can't assert `MagicMock('example file')` == `MagicMock('example file')`!
They have internal IDs and don't do `__eq__` nicely :(((

I set up `MockFile` so that the tests in `test_googleplay` are nicer, but it's kinda feels_bad that we need our own class for this

Reply:
Yeah, I agree, it doesn't look great. I took some time to think about it and maybe the best solution is to let `push_apk()` accept both a file handler (when it's called by `__main__`) and a string (when it's called by pushapkscript). This way, the `push_apk()` call remains simple enough and testing both scenarios in mozapkpublisher doesn't look too hard. 

What do you think?

Reply:
Hmm :thinking:, two thoughts:

* Arguably, pushing this into `mozapkpublisher` doesn't reduce the overall complexity, it just moves logic - we'd just be moving these tests, too
* This would add additional logic to `mozapkpublisher` - if some of the parameters can be either a string _or_ a file, then we need to appropriately open a file and manage closing it.

:thinking:

Reply:
True :thinking:  How about we stop asking for a FileType at https://github.com/mozilla-releng/mozapkpublisher/blob/cceec0f77288a38490e17c450b40f403b5cca4fb/mozapkpublisher/push_apk.py#L184, and we just open files in `push_apk()`?

Reply:
That will definitely address the "additional logic" issue, but I'm still on-the-fence as to what benefit is being provided to push this into mozapkpublisher (tests involving opening files will still be needed, the code is still there, etc).

Also, one downside of having it in `mozapkpublisher` is that we'll lose control: what if we want to call `push_apk` multiple times from `pushapkscript`? We might be reusing some files, such as the Google Play creds. If `mozapkpublisher` controls file handling, then we're forced to keep closing and reopening files between calls.

It's kinda like that Python talk where the guy talked about pulling IO up so that the lower functions could focus on behaviour. Perhaps that argument is applicable here, too?

Reply:
Yeah, closing and reopening files would suck. Okay to keep the current behavior, then :) 

Regarding the python talk, as far as I remember, it was more about dealing with IO at the beginning and processing the extracted raw data, then. In this case, we would just be passing IO handlers down.